### PR TITLE
Update bash script for Mac / Unix

### PR DESCRIPTION
**Pull request by pythonmagic**
Update bash script for Mac / Unix

### CODE REVIEW

**Comment by StevenBlack** [comment]
Hi @pythonmagic thanks for the PR.

First, to clarify everything, is this in addition to #216?  Or should we close #216?

**Comment by pythonmagic** [comment]
i can close #216 and we can use #219 for update , sorry for confusion.

File: updateHosts.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+# This script will create in first running backup of ORIGINAL/CURRENT hosts file in hosts.skel file.
+# If hosts.skel file exists, then NEW copy with customized unified hosts file will be copied to proper path.
+# Next DNS Cache will be refreshed.
+# YOU NEED RUNNING THIS SCRIPT FILE IN COMMAND LINE PROMPT WITH ADMINISTRATOR PRIVILIGES
```
Comment (FadeMind) [inline updateHosts.sh]: Is not needed this warning cause we using `sudo` in script

File: updateHosts.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+#
+# Might have to change change execute for the script file
+# chmod 755 updaeHosts.sh
+#
+# ./updaeHosts.sh OR sh updaeHosts.sh
+
+flushDNS() {
+	# Find OS name
+	#if Darwin (Mac OSX)
+	#if Linux (Ubunut /Debian)
```
Comment (FadeMind) [inline updateHosts.sh]: Typo: `Ubunut`
Correct: `Ubuntu`

**Comment by StevenBlack** [comment]
@pythonmagic can you PLEASE stop creating new pull requests?

Here's how it works:

1. Create a new branch.  
2. Make all changes on that branch.  
3. Submit the PR from that branch.  
4. Later, if you push new changes in the branch, these automatically update the PR here.

Try it, it works  😸

File: updateHosts.sh [hunk 3]
```diff
@@ -50,6 +40,37 @@ updateHosts() {
...
+# Restore default Hosts file.
+restore()
+{
+  OUT=$(defaultHosts)
+  echo "* * * Restoring default hosts * * *"
+  echo "This is restored default hosts file: $OUT" > hosts
+	sudo cp hosts /etc/hosts
+  flushDNS
+}
+
```
Comment (StevenBlack) [inline updateHosts.sh]: We can't presume the contents of a default hosts file.  See Issue #87.

Comment (pythonmagic) [inline updateHosts.sh]: the objective is revert back host file to OS default , one that come of freshly loaded OS.
Similar Ad-away does for android devices , time to time have found it  extremely useful.

i have used template from Microsoft website and cross checked on Mac OSX too.

**Comment by FadeMind** [comment]
@StevenBlack I using this simple code:
```
#!/usr/bin/env bash

cmd_run() {
    ${TERM} -e '/usr/bin/python updateHostsFile.py -a --replace --flush-dns-cache'
}

for TERM in konsole gnome-terminal lxterminal mate-terminal; do
    if [ -f /usr/bin/${TERM} ] ; then cmd_run ; fi
done

```

I using KDE Plasma @ Manjaro, so file launched from Dolphin File Browser will run in new Konsole window and it will be closing after end job. **KISS**

**Comment by ScriptTiger** [comment]
I'm not a Mac/Unix person, but it seems to me this PR has been here since last year and the last comment from Steven was less than positively reinforcing this to get approved. I think the changes proposed in this may have already been worked through by other means, anyway, but I can't say that for sure. Can this be closed?

**Pull request closed without merging**